### PR TITLE
Adding waypoints example

### PR DESCRIPTION
**Pull request by shakthi-prashanth-m**
This example introduces adding waypoint missions.
The original source code for the example is from integration tests.
Couldn't verify  the images after executing TAKE_PHOTO mission.
Need to verify.

### CODE REVIEW

File: example/fly_mission/fly_mission_dronecore.cpp [hunk 1]
```diff
@@ -0,0 +1,357 @@
+//
+// Simple example to demonstrate how to use DroneCore.
```
Comment (julianoes) [inline example/fly_mission/fly_mission_dronecore.cpp]: Maybe adapt this to say `Example to demonstrate how to fly misison using DroneCore`.

File: example/fly_mission/fly_mission_dronecore.cpp [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+//
+// Author: Julian Oes <julian@oes.ch>
```
Comment (julianoes) [inline example/fly_mission/fly_mission_dronecore.cpp]: Add yourself to the list 😉

File: example/takeoff_land/CMakeLists.txt [hunk 3]
```diff
@@ -9,18 +9,19 @@ set(CMAKE_CXX_FLAGS_DEBUG "-DDEBUG")
 
 # This finds thread libs on Linux, Mac, and Windows.
 find_package(Threads REQUIRED)
+message(${CMAKE_SOURCE_DIR})
 
 include_directories(
     # Not needed if installed system-wide
-    ../install/include
+    ${CMAKE_SOURCE_DIR}/../../install/include
```
Comment (julianoes) [inline example/takeoff_land/CMakeLists.txt]: Thanks for doing the rename. However, I'm puzzled how you can start from the root source dir, then go back twice and find `install/include` there? What am I missing?

Comment (shakthi-prashanth-m) [inline example/takeoff_land/CMakeLists.txt]: :) ${CMAKE_SOURCE_DIR} - refers to source directory where CMakeLists.txt resides i.e, `/home/intel/DroneCore/example/takeoff_land`. Hence, going back twice.

File: example/fly_mission/fly_mission_dronecore.cpp [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+                finished = true;
+                break;
+
+            case MissionState::ERROR:
+                finished = true;
+                break;
+
+        }
+
+        std::this_thread::sleep_for(std::chrono::milliseconds(10));
```
Comment (mrpollo) [inline example/fly_mission/fly_mission_dronecore.cpp]: how did you come up with the 10ms delay?

File: example/fly_mission/fly_mission_dronecore.cpp [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+                     10.0f, 2.0f, false,
+                     -90.0f, 30.0f,
+                     MissionItem::CameraAction::STOP_VIDEO));
+
+    mission_items.push_back(
+        add_waypoint(47.398100366082858,
+                     8.5456969141960144,
+                     10.0f, 5.0f, false,
+                     -45.0f, -30.0f,
+                     MissionItem::CameraAction::START_PHOTO_INTERVAL));
```
Comment (julianoes) [inline example/fly_mission/fly_mission_dronecore.cpp]: 1s according to:
https://github.com/dronecore/DroneCore/blob/1f0e92723e6c85de2d2a069ac3d4436dc9b8028e/plugins/mission/mission_impl.cpp#L353
https://github.com/dronecore/DroneCore/blob/5110ab992871e8f4b254295a70d955e9f178efe9/plugins/mission/mission_item_impl.h#L30
https://github.com/dronecore/DroneCore/blob/5110ab992871e8f4b254295a70d955e9f178efe9/plugins/mission/mission_item_impl.h#L53

File: example/fly_mission/fly_mission_dronecore.cpp (hunk 4 above)
Comment (julianoes) [inline example/fly_mission/fly_mission_dronecore.cpp]: This is just how fast to loop through the while, doesn't matter to much.

File: example/fly_mission/fly_mission_dronecore.cpp (hunk 5 above)
Comment (mrpollo) [inline example/fly_mission/fly_mission_dronecore.cpp]: Thanks!

File: example/takeoff_land/CMakeLists.txt (hunk 3 above)
Comment (shakthi-prashanth-m) [reply example/takeoff_land/CMakeLists.txt]: Did you mean links to example in https://github.com/dronecore/DroneCore/blob/master/docs/build.md#build-example ?
Also, link to CMakeLists.txt in the last line is invalid. We shall update both of these links to either of the example: `fly_mission` or `takeoff_land`

Comment (shakthi-prashanth-m) [reply example/takeoff_land/CMakeLists.txt]: I can edit [build.md](https://github.com/dronecore/DroneCore/blob/master/docs/build.md) only after this pull requests get merged. Isn't it ?

Comment (julianoes) [reply example/takeoff_land/CMakeLists.txt]: No, just add the doc change here please.

**Comment by julianoes** [comment]
This PR was replaced by #39.

**Pull request closed without merging**